src/middlewared/middlewared/etc_files/initiators.allow.mako

```diff
@@ -1,7 +1,20 @@
 <%
+    import ipaddress
     base_name = middleware.call_sync('iscsi.global.config')['basename']
     targets = middleware.call_sync('iscsi.target.query', [['auth_networks', '!=', []]])
+
+    def parse_auth(auth):
```

Comment:
I was thinking more like this (instead of doing `auth.split`). This also has benefit of "normalizing" the network entries. i.e. if someone does something like `192.168.1.128/16`, it'll return `192.168.0.0/16`.

```
def parse_auth(auth):
    try:
        ipobj = ipaddress.ip_interface(auth)
    except ValueError:
        middleware.logger.warning("Invalid IP address: %s", auth, exc_info=True)
    else:
        if ipobj.network.prefixlen in (32, 128):
            return str(ipobj.ip)
        return str(ipobj.network)

Reply:
I actually looked at the API and we use the same python module to validate the input so it should be an extreme edge-case if this function fails. (i.e. someone made manual changes to db or we have some ancient legacy version that like 5 people (total) are using).